src/aliceVision/sfmData/colorize.cpp

```diff
@@ -31,97 +62,49 @@ void colorizeTracks(SfMData& sfmData)
-      : viewId(viewId)
-      , cardinal(cardinal)
-    {}
-
-    IndexT viewId;
-    std::size_t cardinal;
-    std::vector<std::reference_wrapper<Landmark>> landmarks;
-  };
-
-  std::vector<ViewInfo> sortedViewsCardinal;
-  sortedViewsCardinal.reserve(sfmData.getViews().size());
-  {
-    // create cardinal per viewId map
-    std::map<IndexT, std::size_t> viewsCardinalMap; // <ViewId, Cardinal>
-    for(const auto& landmarkPair : sfmData.getLandmarks())
-    {
-      const Observations& observations = landmarkPair.second.observations;
-      for(const auto& observationPair : observations)
-        ++viewsCardinalMap[observationPair.first]; // TODO: 0
-    }
-
-    // copy key-value pairs from the map to the vector
-    for(const auto& cardinalPair : viewsCardinalMap)
-      sortedViewsCardinal.push_back(ViewInfo(cardinalPair.first, cardinalPair.second));
-
-    // sort the vector, biggest cardinality first
-    std::sort(sortedViewsCardinal.begin(),
-              sortedViewsCardinal.end(),
-              [] (const ViewInfo& l, const ViewInfo& r) { return l.cardinal > r.cardinal; });
-  }
+  std::vector<LMColorAccumulator> landmarkInfo(sfmData.getLandmarks().size());
+  const Views& views = sfmData.getViews();
 
-  // assign each landmark to a view
-  for(ViewInfo& viewCardinal : sortedViewsCardinal)
+  #pragma omp parallel for
+  for( int viewId = 0; viewId < views.size(); ++viewId)
   {
-    std::vector<std::reference_wrapper<Landmark>> toKeep;
-    const IndexT viewId = viewCardinal.viewId;
+      image::Image<image::RGBColor> image;
+      image::readImage(views.at(viewId)->getImagePath(), image, image::EImageColorSpace::SRGB);
 
-    for(int i = 0; i < remainingLandmarksToColor.size(); ++i)
-    {
-      Landmark& landmark = remainingLandmarksToColor.at(i);
-      auto it = landmark.observations.find(viewId);
-      if(it != landmark.observations.end())
-      {
-        viewCardinal.landmarks.push_back(landmark);
-      }
-      else
+      #pragma omp parallel for
+      for(int i = 0; i < remainingLandmarksToColor.size(); ++i)
       {
-        toKeep.push_back(landmark);
+            const Landmark& landmark = remainingLandmarksToColor.at(i);
+            const auto it = landmark.observations.find(viewId);
+
+            if(it != landmark.observations.end())
+            {
+                const Vec3& Tcenter = sfmData.getAbsolutePose(viewId).getTransform().center();
+                const Vec3& pt = landmark.X;
+                const double eucd = 1.0 / (Tcenter - pt).norm();
+                Vec2 uv = it->second.x;
+                uv.x() = clamp(uv.x(), 0.0, static_cast<double>(image.Width() - 1));
+                uv.y() = clamp(uv.y(), 0.0, static_cast<double>(image.Height() - 1));
+                const image::RGBColor obsColor = image(uv.y(), uv.x());
+                image::RGBfColor& rgbf = image::RGBfColor(static_cast<float>(obsColor.r()), static_cast<float>(obsColor.g()), static_cast<float>(obsColor.b()));
```

Comment:
No reason to have a reference here and no need of explicit conversion for char to float.
```suggestion
                image::RGBfColor rgbf(obsColor.r(), obsColor.g(), obsColor.b());
```

Reply:
Is there no need for explicit conversion of float to char either? I vaguely remember having an overflow problem with implicit conversion between these 2 types somewhere, will test again

Reply:
Yes, float to char needs an explicit conversion. Globally if you convert to a smaller type, you need an explicit conversion.